differs/rpm_diff.go

```diff
@@ -364,3 +366,115 @@ func unlock() error {
 	daemonMutex.Unlock()
 	return nil
 }
+
+type RPMLayerAnalyzer struct {
+}
+
+// Name returns the name of the analyzer.
+func (a RPMLayerAnalyzer) Name() string {
+	return "RPMLayerAnalyzer"
+}
+
+// Diff compares the installed rpm packages of image1 and image2 for each layer
+func (a RPMLayerAnalyzer) Diff(image1, image2 pkgutil.Image) (util.Result, error) {
+	diff, err := singleVersionLayerDiff(image1, image2, a)
+	return diff, err
+}
+
+// Analyze collects information of the installed rpm packages on each layer
+func (a RPMLayerAnalyzer) Analyze(image pkgutil.Image) (util.Result, error) {
+	analysis, err := singleVersionLayerAnalysis(image, a)
+	return analysis, err
+}
+
+// getPackages returns an array of maps of installed rpm packages on each layer
+func (a RPMLayerAnalyzer) getPackages(image pkgutil.Image) ([]map[string]util.PackageInfo, error) {
+	path := image.FSPath
+	var packages []map[string]util.PackageInfo
+	if _, err := os.Stat(path); err != nil {
+		// invalid image directory path
+		return packages, err
+	}
+
+	// try to find the rpm binary in bin/ or usr/bin/
+	rpmBinary := filepath.Join(path, "bin/rpm")
+	if _, err := os.Stat(rpmBinary); err != nil {
+		rpmBinary = filepath.Join(path, "usr/bin/rpm")
+		if _, err = os.Stat(rpmBinary); err != nil {
+			logrus.Errorf("Could not detect RPM binary in unpacked image %s", image.Source)
+			return packages, nil
+		}
+	}
+
+	packages, err := rpmDataFromLayerFS(image)
+	if err != nil {
```

Comment:
Do we care about the error here? This fallback pattern makes sense to me here, but I feel like we might want to be logging this error (I know we're not doing it in `RPMAnalyzer.getPackages()` either). At the very least we should add something to the log message saying something like `Couldn't retrieve RPM data from extracted filesystem; running query in container`.

Reply:
Agree, lets include some logging